core/modules/node/node.types.inc

```diff
@@ -130,6 +130,92 @@ function node_type_form($form, &$form_state, $type = NULL) {
+  }
+
+  $permissions_assigned = FALSE;
+  $content_type_permissions = node_type_form_permissions($type->type);
+  foreach ($content_type_permissions['checkboxes'] as $role => $checkboxes) {
+    if (!empty($checkboxes['#default_value'])) {
+      // At least one permission related to this content type is assigned to at
+      // least one role.
+      $permissions_assigned = TRUE;
+      break;
+    }
+  }
+
+  // Edge case, where both the default admin role and the default content editor
+  // role have been disabled in in admin/config/people/roles, while at the same
+  // time no other role has been granted any permission associated with this
+  // content type. We show this warning for both new and existing content types.
+  if (!$permissions_assigned) {
+    $permissions_message = t('No permissions assigned for this content type. Content of this type may not be able to be created, updated or deleted until permissions have been configured appropriately.');
+    $permissions_message_type = 'warning';
+  }
+  // Show some info messages if automatic permission assignment is enabled for
+  // at least one of the default admin/editor roles. We only show these when
+  // creating a new content type.
+  if (!empty($auto_permission_roles) && empty($type->type)) {
+    $permissions_message = format_plural(
+      count($auto_permission_roles),
+      'The %role role has been assigned permissions to create, edit and delete any content of this type.',
+      'The %admin and %editor roles have been assigned permissions to create, edit and delete any content of this type.',
+      $auto_permission_roles
+    );
+    $permissions_message_type = 'info';
+  }
+
+  if (isset($permissions_message)) {
+    if (user_access('administer permissions')) {
+      $permissions_message .= ' ' . t('You can make any necessary adjustments in the <em>Permissions</em> tab below.');
+    }
+    else {
+      $user_module_permissions = module_invoke('user', 'permission');
+      $permissions_message .= ' ' . t('Only roles with the %permission permission can make any necessary adjustments.', array('%permission' => $user_module_permissions['administer permissions']['title']));
+    }
+
+    // We are mimicking the style of a warning/info message here. If we used
+    // backdrop_set_message(), the message would be placed at the top of the
+    // page, which would make it unnecessarily prominent and potentially
+    // distracting. We are placing it right above the vertical tabs section
+    // instead (closer to where permissions can be configured). Intentionally
+    // not placing it inside the "Permissions" tab either, since that is not the
+    // default tab, so it wouldn't be visible, and we need to show some warnings
+    // to people even if they don't have access to the "Permissions" tab.
+    $form['permissions_message_container'] = array(
+      '#type' => 'container',
+      '#attributes' => array('class' => array('messages', $permissions_message_type)),
+    );
```

Comment:
It would be a nice follow-up to establish a pattern for this, or possibly a new element type similar to `#type => help` (maybe `#type => messages`)? I think we only do this in one or two other places in all of core.

Reply:
We already have https://github.com/backdrop/backdrop-issues/issues/3329 @quicksketch, which is on my radar 👀